### PR TITLE
chore: Add detailed error message for specific exceptions

### PR DESCRIPTION
**Pull request by filzrev**
This PR is intended to help diagnose the CI error which occurred in #9744.

**Error Message**
> /home/runner/work/docfx/docfx/docs/extensions/tools.yml: error InvalidInputFile: Unable to load file 'extensions/tools.yml' via processor 'Dashboard': Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')

It seems to be throwing an `ArgumentOutOfRangeException`. but there is no StackTrace information contained in log.
So I've added full exception message to following exception types.

- `ArgumentException` derived class
- `NullReferenceException`  
   (It occurred at #9146)

### CODE REVIEW

File: src/Docfx.Build/HostServiceCreator.cs [hunk 1]
```diff
@@ -63,8 +63,12 @@ public HostServiceCreator(DocumentBuildContext context)
             {
                 if (e is not DocumentException)
                 {
+                    var message = e is ArgumentException or NullReferenceException
+                        ? e.ToString()
+                        : e.Message;
+
                     Logger.LogError(
-                        $"Unable to load file '{file.File}' via processor '{processor.Name}': {e.Message}",
+                        $"Unable to load file '{file.File}' via processor '{processor.Name}': {message}",
```
Comment (yufeih) [inline src/Docfx.Build/HostServiceCreator.cs:71]: Ah, we should have a `LogError` overload that takes an `Exception` that logs exception message as error and full exception as verbose.

**Comment by filzrev** [comment]
This PR is not expected to have any snapshot diffs.
But percy workflow is pending with the following status.

> percy/docfx Expected — Waiting for status to be reported

**Comment by filzrev** [comment]
Please keep this PR open until `InvalidInputFile` error reproduced.

I suspect that `EnvironmentContext` or `Directory.CurrentDirectory` related problems.
Because `[Collection("docfx STA")]` attribute is not defined for `SamplesTest`.

**Comment by filzrev** [comment]
This error is occurred on my forked repository's CI today.
It seems this issue occurs at a very low rate.

**Comment by filzrev** [comment]
> This error is occurred on my forked repository's CI today.
> It seems this issue occurs at a very low rate.

I've tried to reproduce problem on forked repository by using `workflow_dispatch` event about 100 times.
But this issue can't reproduced.

@yufeih 
Is it able to temporary merge this PR to `main` for capturing detailed logging message purpose?

> Ah, we should have a LogError overload that takes an Exception that logs exception message as error and full exception as verbose.

**Review by yufeih — APPROVED** [review]

**Comment by filzrev** [comment]
It can reproduced on my forked repository.
It seems `JsonSchema.Net` side problems. that occurs when using v6.0.3 or later.
I've reported the issue at https://github.com/gregsdennis/json-everything/issues/693

**Exception logs**
```
/Users/runner/work/docfx/docfx/docs/extensions/tools.yml: error InvalidInputFile: Unable to load file 'extensions/tools.yml' via processor 'Dashboard': System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at System.Linq.Enumerable.TryGetSingle[TSource](IEnumerable`1 source, Boolean& found)
   at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source)
   at Json.Schema.JsonSchema.CheckScopedConstraints(DynamicScope scope)
   at Json.Schema.JsonSchema.BuildConstraint(JsonPointer evaluationPath, JsonPointer baseInstanceLocation, JsonPointer relativeInstanceLocation, DynamicScope scope)
   at Json.Schema.JsonSchema.Evaluate(JsonNode root, EvaluationOptions options)
   at Json.Schema.JsonSchemaExtensions.Evaluate(JsonSchema jsonSchema, JsonElement jsonElement, EvaluationOptions options)
   at Json.Schema.JsonSchemaExtensions.Evaluate(JsonSchema jsonSchema, JsonDocument jsonDocument, EvaluationOptions options)
   at Docfx.Build.SchemaDriven.SchemaValidator.Validate(Object obj) in /_/src/Docfx.Build.SchemaDriven/Validators/SchemaValidator.cs:line 28
   at Docfx.Build.SchemaDriven.SchemaDrivenDocumentProcessor.Load(FileAndType file, ImmutableDictionary`2 metadata) in /_/src/Docfx.Build.SchemaDriven/SchemaDrivenDocumentProcessor.cs:line 114
   at Docfx.Build.Engine.HostServiceCreator.Load(IDocumentProcessor processor, ImmutableDictionary`2 metadata, FileMetadata fileMetadata, FileAndType file) in /_/src/Docfx.Build/HostServiceCreator.cs:line 60
```

**Pull request merged**